src/platform/tls_schannel.c

```diff
@@ -3045,19 +3163,44 @@ QuicHpComputeMask(
     )
 {
     ULONG TempSize = 0;
-    QUIC_STATUS Status =
-        NtStatusToQuicStatus(
-        BCryptEncrypt(
-            (BCRYPT_KEY_HANDLE)Key,
-            (uint8_t*)Cipher,
-            QUIC_HP_SAMPLE_LENGTH * BatchSize,
-            NULL,
-            NULL,
-            0,
-            Mask,
-            QUIC_HP_SAMPLE_LENGTH * BatchSize,
-            &TempSize,
-            0));
+    QUIC_STATUS Status = QUIC_STATUS_SUCCESS;
+    if (Key->Aead == QUIC_AEAD_CHACHA20_POLY1305) {
+        uint8_t Zero[5] = {0};
+        Key->Info->cbNonce = QUIC_HP_SAMPLE_LENGTH;
+        for (uint32_t i = 0, Offset = 0; i < BatchSize; ++i, Offset += QUIC_HP_SAMPLE_LENGTH) {
+            Key->Info->pbNonce = (uint8_t*)(Cipher + Offset);
+            Status =
+                NtStatusToQuicStatus(
+                BCryptEncrypt(
+                    Key->Key,
+                    Zero,
+                    sizeof(Zero),
+                    Key->Info,
+                    NULL,   //Key->Info->pbNonce,
+                    0,      //QUIC_HP_SAMPLE_LENGTH,
+                    Mask + Offset, 
+                    QUIC_HP_SAMPLE_LENGTH, // This might fail because the Tag won't fit...
```

Comment:
Why wouldn't the tag fit?

Reply:
The tag is 16 bytes, and QUIC_HP_SAMPLE_LENGTH is 16 bytes, but the output is length of Mask + length of Tag, which is 21 bytes.
We could solve this by making the sample length larger, and discarding the unused parts on AES.  Ideally, we'd be able to compute ChaCha20 without the Poly1305 tag, and this won't be a problem.